cmd/openfga/openfga_test.go

```diff
@@ -0,0 +1,967 @@
+			},
+		},
+		{
+			name: "invalid type definition (empty type name)",
+			input: &openfgapb.WriteAuthorizationModelRequest{
+				StoreId: testutils.RandomID(t),
+				TypeDefinitions: &openfgapb.TypeDefinitions{
+					TypeDefinitions: []*openfgapb.TypeDefinition{
+						{
+							Type: "",
+							Relations: map[string]*openfgapb.Userset{
+								"viewer": {Userset: &openfgapb.Userset_This{}},
+							},
+						},
+					},
+				},
+			},
+			output: output{
+				errorCode: codes.InvalidArgument,
+			},
+		},
+		{
+			name: "invalid type definition (too many chars in name)",
+			input: &openfgapb.WriteAuthorizationModelRequest{
+				StoreId: testutils.RandomID(t),
+				TypeDefinitions: &openfgapb.TypeDefinitions{
+					TypeDefinitions: []*openfgapb.TypeDefinition{
+						{
+							Type: testutils.CreateRandomString(255),
+							Relations: map[string]*openfgapb.Userset{
+								"viewer": {Userset: &openfgapb.Userset_This{}},
+							},
+						},
+					},
+				},
+			},
+			output: output{
+				errorCode: codes.InvalidArgument,
+			},
+		},
+		{
+			name: "invalid type definition (invalid chars in name)",
+			input: &openfgapb.WriteAuthorizationModelRequest{
+				StoreId: testutils.RandomID(t),
+				TypeDefinitions: &openfgapb.TypeDefinitions{
+					TypeDefinitions: []*openfgapb.TypeDefinition{
+						{
+							Type: "some type",
+							Relations: map[string]*openfgapb.Userset{
+								"viewer": {Userset: &openfgapb.Userset_This{}},
+							},
+						},
+					},
+				},
+			},
+			output: output{
+				errorCode: codes.InvalidArgument,
+			},
+		},
+	}
+
+	conn := connect(t, tester)
+	defer conn.Close()
+
+	client := openfgapb.NewOpenFGAServiceClient(conn)
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			response, err := client.WriteAuthorizationModel(context.Background(), test.input)
+
+			s, ok := status.FromError(err)
+			require.True(t, ok)
+			require.Equal(t, test.output.errorCode.String(), s.Code().String())
+
+			if test.output.errorCode == codes.OK {
+				_, err = id.Parse(response.AuthorizationModelId)
+				require.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestFunctionalHTTP(t *testing.T) {
+
+	require := require.New(t)
+
+	tester, err := newOpenFGATester(t)
+	require.NoError(err)
+	defer tester.Cleanup()
+
+	t.Run("TestCreateStore", HTTPCreateStoreTest)
+	t.Run("TestListStores", HTTPListStoresTest)
+	t.Run("TestDeleteStore", HTTPDeleteStoreTest)
+	t.Run("TestWrite", HTTPWriteTest)
+	t.Run("TestRead", HTTPReadTest)
+	t.Run("TestCheck", HTTPCheckTest)
+	t.Run("TestExpand", HTTPExpandTest)
+}
+
+func HTTPCreateStoreTest(t *testing.T) {
+
+}
+
+func HTTPListStoresTest(t *testing.T) {
+
+}
+
+func HTTPDeleteStoreTest(t *testing.T) {
+
+}
+
+func HTTPWriteTest(t *testing.T) {
+
+}
+
+func HTTPReadTest(t *testing.T) {
+
+}
+
+func HTTPCheckTest(t *testing.T) {
+
+}
+
+func HTTPExpandTest(t *testing.T) {
+
+}
+
+func tuple(object, relation, user string) *openfgapb.TupleKey {
```

Comment:
I believe a util was recently introduced to this.

Reply:
Oops. That's residual. It's in the `tuple` package.